### ISSUE TITLE
Can't download a resource

### ISSUE THREAD

**Issue by saisiddu** [hydroshare/hydroshare#2347]
Resource: https://www.hydroshare.org/resource/20ff6a71b0ef4970b74cc7a53a523c4f/  
  - it's currently discoverable (not Public). 

Error on Welcome Page:
![image](https://user-images.githubusercontent.com/7194053/31527471-4d3c5bbe-af82-11e7-9350-6136a4fc7a9b.png)

My contact: saisiddu@uw.edu

**Comment by dtarb** [comment]
@saisiddu does the user attempting this have file access permissions over the resource.  You mention it being discoverable but do not indicate if the user attempting this is you (the owner) who should be able to download files, or someone else.  It is correct for access to files for a discoverable resource to be blocked for a user who has not had the resource explicitly shared with them.  

@ChristinaB  may be able to help as she is familiar with HydroShare and this seems to be a landlab related resource.  She is at UW.

Also @Castronova can you look into this and check whether access control is working as it should through the REST client from Jupyterhub issue.  Please investigate whether there is a way to give a clearer error message for resources properly blocked by access control.

**Comment by saisiddu** [comment]
@dtarb: Yes, I was trying to access my resource. This error doesn't occur with any of my other resources. I will talk to @ChristinaB and @Castronova.
Thank you!

**Comment by dtarb** [comment]
What is different about this resource.  I suggest you share it with me (dtarb) and Tony (TonyCastronova) so we can see if we can see what is causing trouble with this one but not the others.

**Comment by saisiddu** [comment]
@dtarb: I have added you and Tony to the resource (https://www.hydroshare.org/resource/20ff6a71b0ef4970b74cc7a53a523c4f/). I have used this resource earlier without any glitches. I don't remember changing anything.  Thank you for looking at this issue.

**Comment by ChristinaB** [comment]
@alvacouch  - thanks for looking at this one

**Comment by ChristinaB** [comment]
Do we need a new resource or does it have a clean bill of health?

**Comment by hyi** [comment]
@ChristinaB @alvacouch @dtarb I have fixed this resource, it can now be downloaded successfully. The issue was with the bag_modified flag, it was set to false, but there was no bag.  Once I manually set it to true, everything just works.

**Comment by hyi** [comment]
@saisiddu @ChristinaB You can probably close this issue after verifying download of this resource now works, even though we have no idea how the bag_modified flag was out of sync with what it should really be, probably resulting from some exception or testing or whatever it was we did not know...

**Comment by saisiddu** [comment]
Thank you @hyi!

**Comment by dtarb** [comment]
@hyi this is a one off fix, not a systematic fix.  I suggest we need a systematic fix.  Could the code that attempts to download a bag catch when there is no bag and then instead of giving an error, set the bag_modified code to true and restart the process so that a bag is generated.  If we had code like this, then if a resource is ever in this state again it will be fixed by the system.

**Comment by hyi** [comment]
@dtarb Good suggestion. I can work on this if you create a new issue for this. It should not take too much time. OK. I see you just created a new issue and assigned it to me. Should be able to get it in for release 1.14.
